server/src/controllers/company.ts

```diff
@@ -20,7 +20,7 @@ export async function getAllCompanies(
 
   const user = await User.findOne({ emailAddress: req.user?.email }).lean().select("isAdmin");
   if (!user) return next(new NotFound("User not found"));
-  if (!user.isAdmin) return next(new Forbidden("Only admins may get all companies."));
+  //if (!user.isAdmin) return next(new Forbidden("Only admins may get all companies."));
```

Comment:
Wenn wir die Liste so ans Frontend senden, dann können Studis einfach sämtliche Firmeninfos auslesen. Soll das so? Es gab ja extra ein Limit von 12 Ergebnissen für die Suche, damit man nicht einfach alle Infos bekommen kann.

Wir bräuchten hier dann einen neuen Endpoint, der die Firmen vorab nach Search Term filtert und nur die Infos, die wirklich benötigt werden, zurückgibt, würde ich sagen. Obwohl sich natürlich bei so einer Funktionalität überhaupt nicht vermeiden lässt, dass mindestens alle Firmennamen abrufbar sind.

Alternativ könnte nur nach vollständigen Firmennamen gesucht werden (also wenn's ein 1:1 Match ist und kein teilweiser). Dann wär es aber weniger responsiv.

Reply:
Wäre es eine Lösung, wenn wir nur die Id und den Namen ans Frontend anfordern? Ich stell mir die Frage, wie wahrscheinlich es ist, dass Studis die Response sich anschauen & dann was mit den Ergebnissen anzufangen wissen – also ich finde es lässt sich streiten, welchen Vorteil Studis haben davon.

Bei der anderen Variante, wo man erst beim Tippen eines Namens nach genau der Company sucht: das ist doch mit ziemlicher Latenz verbunden, oder? Wenn die Verbindung gerade nicht so toll ist, dann dauert es doch mitunter, bis ein Ergebnis vorgeschlagen wird, und auch während des Tippens sollen nicht mehrere Requests ans Backend gehen.

Mir fällt sonst echt grad keine passende Alternative ein ...

Reply:
Eine Idee, find ich aber so naja: man könnte, wenn man auf Speichern klickt, checken ob es eine ähnliche Firma gibt, und dann sagen: Hey, meintest du vielleicht diese hier? So ähnlich wie wir es machen, wenn es eine neue Firma ist, zu der es noch keine Adresse gibt.

Reply:
also refers to: #163

Reply:
oder im Prinzip wie hier schon vorgeschlagen :D #54

Reply:
> Eine Idee: man könnte, wenn man auf Speichern klickt, checken ob es eine ähnliche Firma gibt, und dann sagen: Hey, meintest du vielleicht diese hier? So ähnlich wie wir es machen, wenn es eine neue Firma ist, zu der es noch keine Adresse gibt. Dann braucht man nicht vorher eine Liste mit Companies und auch Companies mit kürzeren Namen können gefunden werden.

Ja, das ist denke ich die beste Lösung 👍